src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -171,3 +171,35 @@ IntDoubleStr ToNumericValueGetter::operator()(
     [[maybe_unused]] const EvaluationContext* context) const {
   return std::string(asStringViewUnsafe(s.getContent()));
 }
+
+// ____________________________________________________________________________
+LiteralOrString makeDatatypeValueGetter::operator()(
```

Comment:
I think the more consistent way to do this, is by solving the COMPLETE expression inside the ValueGetter,
s.t. the ValueGetter directly returns the IRI.
Your current design has some cases handled in the ValueGetter and some in the expression, which I find a little bit strange.
It actually happens pretty often that we can solve unary expressions directly inside the value getter.